src/database/queries/agendaItems.ts

```diff
@@ -120,12 +120,20 @@ export const insertAgendaItems = async (
 export const insertAgendaItemSubjectTerms = async (
   db: Kysely<DB>,
   items: AgendaItemSubjectTerm[],
-) => {
-  await db.transaction().execute(async (trx) => {
-    await trx
+): Promise<number> => {
+  const uniqueRows = Array.from(
+    items
+      .reduce((map, obj) => {
+        const key = `${obj.agendaItemId}::${obj.subjectTermSlug}`;
+        return map.set(key, obj);
+      }, new Map())
+      .values(),
+  );
```

Comment:
With current approach the type info of unique rows is lost as the Map is declared without generics.
It is possible to add explicit generics to fix that...
<img width="598" height="215" alt="Screenshot 2025-07-15 at 9 02 15 PM" src="https://github.com/user-attachments/assets/2f9e7020-4f2f-4e26-9865-839fc8e21c67" />


But if we instead use the Map's iterator constructor it can infer the types automatically
```suggestion
  const uniqueRows = [
    ...new Map(
      items.map((row) => [`${row.agendaItemId}::${row.subjectTermSlug}`, row]),
    ).values(),
  ];
```